pkg/server/desthost_backend_manager_test.go

```diff
@@ -0,0 +1,102 @@
+package server
+
+import (
+	"context"
+	"testing"
+
+	"sigs.k8s.io/apiserver-network-proxy/proto/agent"
+)
+
+type fakeAgent struct {
+	ctx context.Context
+	agent.AgentService_ConnectServer
+}
+
+func (fa *fakeAgent) Context() context.Context {
+	return fa.ctx
+}
+
+const contextNameKey key = iota
+
+func newNamedBackend(name string) *backend {
+	return &backend{
+		conn: &fakeAgent{
+			ctx: context.WithValue(context.Background(), contextNameKey, name),
+		},
+	}
+}
+
+func TestBackend(t *testing.T) {
+
+	testCases := []struct {
+		name                string
+		destHost            string
+		backends            map[string][]*backend
+		expectedBackendName string
+	}{
+		{
+			name:     "Literal match",
+			destHost: "sts.amazonaws.com",
+			backends: map[string][]*backend{
+				"sts.amazonaws.com": {newNamedBackend("sts.amazonaws.com")},
+			},
+			expectedBackendName: "sts.amazonaws.com",
+		},
+		{
+			name:     "Wildcard match",
+			destHost: "sts.amazonaws.com",
+			backends: map[string][]*backend{
+				".amazonaws.com": {newNamedBackend(".amazonaws.com")},
```

Comment:
why support both .amazonaws.com and *.amazonaws.com?

If we only support the latter, it seems more clear and simpler.

Reply:
Any reason you chose leading dot instead of leading asterisk? The asterisk seems closer to DNS spec (https://datatracker.ietf.org/doc/html/rfc4592#section-2.1.1). I don't see a lot of precedence for leading dots indicating wildcard.

Reply:
@jkh52 These vars are generally not standardized, so the effective standard that exist is curls behavior and curl only works with dot and not with an asterisk:
```bash
# Set proxy
$ export HTTPS_PROXY=http://undef.com

# proxy is invalid, so by default this doesn't work
$ curl https://google.com -v
* Uses proxy env variable HTTPS_PROXY == 'http://undef.com'
*   Trying 209.59.192.38:1080...
* connect to 209.59.192.38 port 1080 failed: Connection refused
* Failed to connect to undef.com port 1080 after 261 ms: Connection refused
* Closing connection 0
curl: (7) Failed to connect to undef.com port 1080 after 261 ms: Connection refused

# Set no proxy for tld with leading dot, works
$ NO_PROXY=.com curl https://google.com -v
* Uses proxy env variable NO_PROXY == '.com'
*   Trying 2a00:1450:4005:80b::200e:443...
* Connected to google.com (2a00:1450:4005:80b::200e) port 443 (#0)

# Set no proxy for tld with leading asterisk, doesn't work
NO_PROXY='*.com' curl https://google.com -v
* Uses proxy env variable NO_PROXY == '*.com'
* Uses proxy env variable HTTPS_PROXY == 'http://undef.com'
*   Trying 209.59.192.38:1080...
* connect to 209.59.192.38 port 1080 failed: Connection refused
* Failed to connect to undef.com port 1080 after 190 ms: Connection refused
* Closing connection 0
curl: (7) Failed to connect to undef.com port 1080 after 190 ms: Connection refused
```